azavea/climate-change-components
# Add ChartsModule

## Overview

Moves a bunch of components that existed in the Lab to this repository so they can be reused elsewhere.

### Demo

No visible changes, app should continue to work as before.


### Notes

The changes in azavea/climate-change-lab#292 to remove lodash led to a very odd problem where `Array.prototype.includes` registers as being unavailable to the components typescript compiler but not the lab's. As far as I can tell, both environments are the same, e.g. same TS, TSC and TSLint versions, so I have no idea why one gets very mad and the other is chill. Rather than dig deeper I just reverted all uses of `includes(x)` to `indexOf(x) >=0`.

I thought about attempting to remove jQuery since its only used by LineGraphComponent and only for a few class selectors and to get the width of some elements. But, jQuery 3 avoids some legacy bloat and handles browser edge cases, so it seemed simpler and cleaner to just leave it in.

I left CurlCopyComponent in Lab based on a discussion on slack. This led to not moving the ChartComponent out of the Lab either. But, it might be nice to have a more comprehensive ChartComponent available for use in other applications since the year slider is in ChartComponent but its also not a separate component by itself. So, I left things as is for now 
to avoid cutting up the existing components even further. The current set in this PR are a good basic set to build other user interfaces. One way to address this would be to create a component in this repo that encapsulates the features of the Year Slider but there are others. Thoughts?

## Testing Instructions

Checkout this branch. Run:
```
yarn
yarn run build:library && npm pack
```

Checkout `feature/awf/chart-components-to-lib#283` in the climate-change-lab.

In that repository, run:
```
yarn
npm install ../climate-change-components/climate-change-components-0.1.0.tgz && yarn run serve
```

Then poke around the lab. Everything should work as before.

Connects azavea/climate-change-lab#283
